webpack, babel config 관련 세팅 (cra eject OR craco)

akswnd98 opened the pull request:
- path alias 설정 (src, src/assets, src/assets/icons, src/assets/svgs, src/assets/theme, * 이상하게 @theme만 잘 안됩니다..)
- typeRoots 설정 (types에 했습니다.)
- /** @jsxImportSource @emotion/react */ 자동 추가 (@emotion/react/css 사용 시 매번 선언해 줘야 하는 문제)

akswnd98 committed:
feat: craco 관련 preset

- path alias 설정 (src, src/assets, src/assets/icons, src/assets/svgs, src/assets/theme, * 이상하게 @theme만 잘 안됩니다..))
- typeRoots 설정 (types에 했습니다.)
- /** @jsxImportSource @emotion/react */ 자동 추가 (@emotion/react/css 사용 시 매번 선언해 줘야 하는 문제)

Roharui commented on craco.config.js:
```diff
@@ -0,0 +1,23 @@
+      '@src/*': path.resolve(__dirname, 'src'),
+      '@src/': path.resolve(__dirname, 'src'),
+      '@assets/*': path.resolve(__dirname, 'src/assets'),
+      '@assets': path.resolve(__dirname, 'src/assets'),
```
`'@assets': path.resolve(__dirname, 'src/assets'),`의 `@assets` <-에 `/`가 없는 이유가 따로 있나요?

akswnd98 replied:
아이고... 어쩐지 안되더라...

akswnd98 replied:
실수에요 ㅎㅎ
근데 저게 @theme 안되는거 원인인줄 알았는데 수정해도 안되네요... ㅜㅜ

akswnd98 replied:
원인을 찾았습니다...
@assests만 정상이었고 나머지가 이상했었네요..
리뷰 감사합니다!

Roharui commented on package.json:
```diff
@@ -3,6 +3,8 @@
   "private": true,
   "dependencies": {
+    "@craco/craco": "^6.4.5",
+    "@emotion/babel-preset-css-prop": "^11.10.0",
```
`@emotion/babel-preset-css-prop`는 `devDependencies`에 있어야 하지 않나요?

akswnd98 replied:
아 맞네요. compile 관련한 부분이니.

Roharui commented on craco.config.js:
```diff
@@ -0,0 +1,23 @@
+  },
+  babel: {
+    presets: [
+      '@emotion/babel-preset-css-prop',
```
`prettier`에 어긋나네요. 수정 부탁드립니다.

akswnd98 committed:
fix: issue 반영

- craco.config.js prettier 반영
- @emotion/babel-preset-css-prop devDependencies로 이동

akswnd98 committed:
fix: path alias 안되는 문제

@theme만 안되는게 아니었습니다.

dalcon10028 reviewed (approved):
고생하셨습니다 :)
